### ISSUE TITLE
Linking Error with Boost Library during Compilation

### ISSUE THREAD

**Issue by bingyang1132** [kahypar/kahypar#185]
Hello,

I encountered a linking error when trying to compile the project on my system. Here are the details:

Error Message:

```
[  0%] Building CXX object kahypar/application/CMakeFiles/KaHyPar.dir/kahypar.cc.o
[  1%] Linking CXX executable KaHyPar
/home/bingyangliu/anaconda3/envs/paprika/lib/libboost_program_options.so.1.82.0: undefined reference to `std::__throw_bad_array_new_length()@GLIBCXX_3.4.29'
collect2: error: ld returned 1 exit status
kahypar/application/CMakeFiles/KaHyPar.dir/build.make:84: recipe for target 'kahypar/application/KaHyPar' failed
make[2]: *** [kahypar/application/KaHyPar] Error 1
CMakeFiles/Makefile2:1236: recipe for target 'kahypar/application/CMakeFiles/KaHyPar.dir/all' failed
make[1]: *** [kahypar/application/CMakeFiles/KaHyPar.dir/all] Error 2
Makefile:140: recipe for target 'all' failed
make: *** [all] Error 2
Environment Details:
```

Operating System: Ubuntu 18.04
Compiler: g++ 7.5.0
Boost version: 1.82.0 (from Anaconda environment: paprika)
I've checked that the symbol GLIBCXX_3.4.29 is present in my libstdc++.so.6. However, I am unable to resolve the linking error. Any assistance or guidance would be greatly appreciated.

Thank you.

**Comment by bingyang1132** [comment]
After further investigation and diagnosis, I found that the root cause of the problem was the version of g++ that I was using(9.0+needed，in README.md).

Upon updating my g++ to a more recent version, the linking error was resolved and I was able to successfully compile the project. I wanted to share this resolution in case someone else encounters a similar issue in the future. It might be helpful to note that ensuring compatibility between the compiler version and required libraries is essential(Since no warning about g++ version is provided).
